syntax/go.vim

```diff
@@ -291,7 +291,7 @@ hi def link     goFunctionCall      Type
 
 " Fields;
 if go#config#HighlightFields()
-  syn match goField                 /\.\w\+\([.\ \n\r\:\)\[,]\)\@=/hs=s+1
+  syn match goField                 /\.\w\+\%([}\!=\><\^|&%\/*\-\+.\ \n\r\:\)\[\],]\)\@=/hs=s+1
```

Comment:
This regular expression is now very hard to understand and/or maintain. It should probably documented better and split over multiple lines.

For example: https://github.com/fatih/vim-go/blob/b81a447a20c59e44fbbcd39e0f8c8c6e1a289afe/syntax/go.vim#L108-L120

(which is still not great, but better than this).

It may also be worth adding `\v` at the start of this pattern for readability.

Reply:
That makes sense. I can clean this regex along with updating `\(` with `\%(`. Sorry for having stepped away from this issue for such a long time by the way.